### PR TITLE
Document how to update documentation

### PR DESCRIPTION
**Pull request by musale**
## Description
I have bootstrapped the steps needed for local installation, building and editing of the documentation for this project.

## Reference to official issue
This addresses the issue #41 

## Motivation and Context
This change enables developers to be able to edit the documentation locally. Since docker is the official development configuration, I used it to create the application and link it to the current development.

## How Has This Been Tested?
Locally, when you run

`$ docker-compose -f local.yml up`

the documentation application will be running on (http://localhost:8002/)[http://localhost:8002/]

## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

**Comment by reubano** [comment]
Hi @musale the PR looks good! I'll follow your instructions and see if I can successfully update the docs.

**Comment by musale** [comment]
@reubano great! Kindly let me know of any changes that would improve the experience

**Comment by lamby** [comment]
@musale This is great, thank you.

**Comment by reubano** [comment]
I got as far as this....

```bash
(alcf) reubano@tokpro ⚡ docker-compose -f local.yml build
postgres uses an image, skipping
Building compile_docs
(alcf) reubano@tokpro ⚡ docker-compose -f local.yml up
Creating network "alcfprod_default" with the default driver
ERROR: failed to update bridge store for object type *bridge.networkConfiguration: open /mnt/sda1/var/lib/docker/network/files/local-kv.db: no such file or directory
```

**Comment by reubano** [comment]
Ok, [getting further](https://github.com/docker/toolbox/issues/317#issuecomment-160478347)...

```bash
(alcf) reubano@tokpro ⚡ docker docker-machine rm default
(alcf) reubano@tokpro ⚡ docker docker-machine create --driver virtualbox default
(alcf) reubano@tokpro ⚡ docker docker-compose -f local.yml up
Creating network "alcfprod_default" with the default driver
Creating volume "alcfprod_postgres_data_dev" with default driver
Pulling postgres (postgres:9.6)...
9.6: Pulling from library/postgres
ad74af05f5a2: Pulling fs layer
...
```

**Comment by musale** [comment]
hi @reubano, did you manage to get your docker up and running?

**Comment by reubano** [comment]
not quite yet...

```bash
(alcf) reubano@tokpro ⚡ docker-compose -f local.yml up
Starting alcfprod_postgres_1 ...
Starting alcfprod_compile_docs_1 ...
Starting alcfprod_postgres_1
Starting alcfprod_prediction_1 ...
Starting alcfprod_compile_docs_1
Starting alcfprod_compile_docs_1 ... done
Starting alcfprod_documentation_1 ...
Starting alcfprod_prediction_1 ... done
Starting alcfprod_interface_1 ...
Starting alcfprod_interface_1 ... done
Attaching to alcfprod_postgres_1, alcfprod_compile_docs_1, alcfprod_prediction_1, alcfprod_documentation_1, alcfprod_interface_1
postgres_1       | LOG:  database system was shut down at 2017-08-22 07:58:20 UTC
compile_docs_1   | make: Entering directory '/app/docs'
compile_docs_1   | make: Leaving directory '/app/docs'
postgres_1       | LOG:  MultiXact member wraparound protections are now enabled
compile_docs_1   | make: *** No rule to make target 'html'.  Stop.
postgres_1       | LOG:  database system is ready to accept connections
alcfprod_compile_docs_1 exited with code 2
interface_1      | Postgres is up - continuing...
interface_1      | python: can't open file 'manage.py': [Errno 2] No such file or directory
interface_1      | python: can't open file 'manage.py': [Errno 2] No such file or directory
alcfprod_interface_1 exited with code 2
prediction_1     |  * Serving Flask app "src/factory.py"
prediction_1     |  * Forcing debug mode on
compile_docs_1   | make: *** /app/docs: No such file or directory.  Stop.
compile_docs_1   | make: *** /app/docs: No such file or directory.  Stop.
compile_docs_1   | make: *** No rule to make target 'html'.  Stop.
```

trying the master branch now to see if I get the same errors...

**Comment by lamby** [comment]
@reubano Can't reproduce your failure. I wonder what's up...

**Comment by reubano** [comment]
@lamby I suspect I have deeper issues related to my docker setup. So I wouldn't worry too much about the above error.

**Comment by reubano** [comment]
@musale I guess this is good to go now. Can you rebase to master and squash to one commit? Thanks!

**Comment by lamby** [comment]
> @musale I guess this is good to go now. Can you rebase to master and squash to one commit? Thanks!

Hey @musale, how are you getting on? :)

File: docs/Makefile [hunk 1]
```diff
@@ -12,6 +12,12 @@ APIDOCDIR_PREDICTION   = apidoc_prediction
 SPHINXAPIDOC  = sphinx-apidoc
 PROJDIR       = ../
 
+# Internal variables.
+ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
+
+# Internal variables.
+ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
```
Comment (WGierke) [inline docs/Makefile]: I guess this happened during merging at some point? :)

Comment (musale) [reply docs/Makefile]: Yes @WGierke I am having a difficult time making this work 😭

**Comment by lamby** [comment]
Just as a drive-by comment for next time, my experience is that whenever your commit message is a bulleted list of 'n' things that it changes, it should have been 'n' commits to begin with — so much easier to review for all involved!

**Comment by musale** [comment]
> Hey @musale, how are you getting on? :)

Hi @reubano and @lamby! I've been offline the past few days. I have come to a host of additional commits so I had some challenges rebasing and squashing. I have failing tests for both documentation and prediction. Can you check if it's good to go on your end?

**Comment by reubano** [comment]
@musale... hmmm, one option (if it is feasible) is to save your updates to a location outside of the git repo. Then do a `git fetch` and `git reset --hard master`. That will get you a clean slate to start from. Then reapply the previous changes. *Be careful though* `git reset --hard` will erase all of your work, so make sure those changes are saved elsewhere.

**Comment by reubano** [comment]
Then once it's looking good, you can `git push -f` back to your PR branch.

**Comment by musale** [comment]
@reubano let me try this. I will update as soon as I'm done

**Comment by musale** [comment]
Hi @reubano, I followed your recommendations and finally managed to fix this PR. Kindly check on your end 😀

**Pull request merged**